vm/src/types/vm_address.rs

```diff
@@ -111,3 +129,22 @@ impl VMAddress {
             .all(|item| item == &0u8)
     }
 }
+
+
+#[cfg(test)]
+mod tests {
+    use crate::{display_util::address_hex, types::VMAddress};
+
+    #[test]
+    fn generate_mock_address_test() {
+        let creator_address = VMAddress::new([
+            111, 119, 110, 101, 114, 95, 95, 95, 95, 95, 95, 95, 95, 95, 95, 95, 95, 95, 95, 95,
+            95, 95, 95, 95, 95, 95, 95, 95, 95, 95, 95, 95,
+        ]);
+        let mock_address = VMAddress::generate_mock_address(&creator_address.to_vec(), 1u64);
+        assert_eq!(
+            address_hex(&mock_address),
+            "0x11111111111111110f0f111111116f776e65725f5f5f5f5f5f5f5f5f5f015f5f"
```

Comment:
This is not correct. The first 10 bytes are 0.
Example generated address from VM:
`0x00000000000000000f0f11111111a94f5374fce5edbc8e2a8697c15331001234`